src/loss.py

```diff
@@ -1,2 +1,32 @@
 from abc import ABC
+import numpy as np
 
+
+class Purity_function(ABC):
+    def __init__(self):
+        pass
+
+    def purity_value(self, label_values: np.array, mask) -> float:
+        pass
+
+
+class Information_gaing(Purity_function):
+    super().__init__()
+
+
+
+    def _entropy(self, label_values: np.array) -> float:
+        _, uniqueClassesCounts = np.unique(label_values, return_counts=True)
+        probabilities = uniqueClassesCounts / uniqueClassesCounts.sum()
+        return sum(probabilities * -np.log2(probabilities))
+
+    def purity_value(self, labels_values: np.array, mask) -> float:
+        a = sum(mask)
```

Comment:
Without looking further not sure what a and b represent - consider renaming variables for clarity